Mage.Sets/src/mage/cards/t/TimotharBaronOfBats.java

```diff
@@ -0,0 +1,147 @@
+    }
+
+    public TimotharBaronOfBats(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{4}{B}{B}");
+
+        this.addSuperType(SuperType.LEGENDARY);
+        this.addSubType(SubType.VAMPIRE);
+        this.addSubType(SubType.NOBLE);
+
+        this.power = new MageInt(4);
+        this.toughness = new MageInt(4);
+
+        // Ward—Discard a card.
+        this.addAbility(new WardAbility(new DiscardCardCost()));
+
+        // Whenever another nontoken Vampire you control dies, you may pay {1} and exile it.
+        // If you do, create a 1/1 black Bat creature token with flying.
+        // It gains “When this creature deals combat damage to a player,
+        // sacrifice it and return the exiled card to the battlefield tapped.”
+        this.addAbility(new DiesCreatureTriggeredAbility(
+                new DoIfCostPaid(new TimotharBaronOfBatsCreateBatEffect(), new ManaCostsImpl<>("{1}"))
+                        .setText("you may pay {1} and exile it. " +
+                                "If you do, create a 1/1 black Bat creature token with flying. " +
+                                "It gains \"When this creature deals combat damage to a player, " +
+                                "sacrifice it and return the exiled card to the battlefield tapped\"."),
+                false,
+                anotherVampireFilter,
+                true
+        ));
+    }
+
+    private TimotharBaronOfBats(final TimotharBaronOfBats card) { super(card); }
+
+    @Override
+    public Card copy() { return new TimotharBaronOfBats(this); }
+}
+
+class TimotharBaronOfBatsCreateBatEffect extends OneShotEffect {
+
+    TimotharBaronOfBatsCreateBatEffect() {
+        super(Outcome.Benefit);
+        staticText = "exile it. " +
+                "If you do, create a 1/1 black Bat creature token with flying. " +
+                "It gains \"When this creature deals combat damage to a player, " +
+                "sacrifice it and return the exiled card to the battlefield tapped\".";
+    }
+
+    private TimotharBaronOfBatsCreateBatEffect(final TimotharBaronOfBatsCreateBatEffect effect) { super(effect); }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player player = game.getPlayer(source.getControllerId());
+        if (player == null) { return false; }
+
+        Card card = game.getCard(targetPointer.getFirst(game, source));
+        if (card == null) { return false; }
+
+        BatToken bat = new BatToken();
+
+        // Create ability to sack the bat and return the vampire
+        TriggeredAbility ability = new DealsCombatDamageToAPlayerTriggeredAbility(
+                new TimotharBaronOfBatsReturnEffect(card, game),
+                false);
+        ability.addEffect(new SacrificeSourceEffect());
+
+        // Turn into a gain ability effect and point it at the bat
+        ContinuousEffect effect = new GainAbilityTargetEffect(ability, Duration.Custom);
```

Comment:
If you're going to use the GainAbility effect, it needs to be done after you create the token.  And the target pointer needs to be set to the Permanent object that was just created.

The way you had it before is probably easier.  I was just pointing out I didn't know if it was correct or not.

Reply:
Fair enough. I'm going to switch back to previous approach.